cmd/endpoint-ellipses.go

```diff
@@ -66,21 +66,19 @@ var isValidSetSize = func(count uint64) bool {
 }
 
 func commonSetDriveCount(divisibleSize uint64, setCounts []uint64) (setSize uint64) {
-	// prefers setCounts to be sorted for optimal behavior.
-	if divisibleSize < setCounts[len(setCounts)-1] {
-		return divisibleSize
-	}
+	// Not necessarily needed but it ensures to the readers
+	// eyes that we prefer a sorted setCount slice for the
+	// subsequent function to figure out the right common
+	// divisor, it avoids loops.
+	sort.Slice(setCounts, func(i, j int) bool {
+		return setCounts[i] > setCounts[j]
+	})
 
 	// Figure out largest value of total_drives_in_erasure_set which results
 	// in least number of total_drives/total_drives_erasure_set ratio.
-	prevD := divisibleSize / setCounts[0]
 	for _, cnt := range setCounts {
 		if divisibleSize%cnt == 0 {
-			d := divisibleSize / cnt
-			if d <= prevD {
-				prevD = d
-				setSize = cnt
-			}
+			return cnt
 		}
```

Comment:
we need to go through the entire list of setCounts() @shichanglin5 - to know the optimal setSize - there may be many erasure ratios possible but we need to choose the best. 
 
Your code will invariably choose something wrong with the sorted setCount - the idea behind commonSetDriveCount() is to get the best possible setCount as well not just the "first" one.

Reply:
I noticed the comments:
```
// Figure out largest value of total_drives_in_erasure_set which results
// in least number of total_drives/total_drives_erasure_set ratio
```
After sorting setCounts from large to small, traverse forward, as long as `divisibleSize%cnt` is satisfied, then current `cnt` is the `largest value of total_drives_in_erasure_set` as mentioned in the comments

Or suppose the value of `setCounts` is [2, 4, 6, 8], can you give an example under what circumstances the same `divisibleSize` returns different values?